src/components/Markdown/MarkdownProvider.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { FC } from 'react';
+import { MDXProvider } from '@mdx-js/react';
+import { HtmlAttributes } from 'src/components/html-component-props';
```

Comment:
I noticed that you used `HTMLAttributes` for defining props in this component, which is causing a TypeScript error. According to the MDX-JS  [documentation](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/mdx/types.d.ts#L22) it appears that using `JSX.IntrinsicElements['h1']` is the recommended approach.

The correct way to define props for the `h1` element in @mdx-js is to use `JSX.IntrinsicElements['h1']`. This ensures that TypeScript recognizes the correct prop types for HTML elements and provides better type safety.

Here's how you can update the code:
```tsx
const H1: FC<JSX.IntrinsicElements['h1']> = ({ children, ...props }) => (
  <h1 className="ui-text-h1 my-40" {...props}>
    {children}
  </h1>
);